circuit_maintenance_parser/__init__.py

```diff
@@ -73,30 +75,29 @@ def get_provider_class(provider_name: str) -> Type[GenericProvider]:
     """Returns the Provider parser class for a specific provider_type."""
     provider_name = provider_name.lower()
 
-    for provider_parser in SUPPORTED_PROVIDERS:
-        if provider_parser.get_provider_type() == provider_name:
+    for provider in SUPPORTED_PROVIDERS:
+        if provider().get_provider_type() == provider_name:
```

Comment:
I'll admit I don't quite understand how this feature is supposed to work/be used. We have the option to override a provider's `provider_type` at init time, but when we're looking up a provider to initialize, we have to use its base (non-overridden) `provider_type`? What does the overridden value get used for, and what does the base value get used for - are there places where they're in conflict? Can I override a provider's `provider_type` to match that of another existing base provider class, (`init_provider(provider_type="ntt", extended_data={"provider_type": "packetfabric"})`) and does that totally break things if so?

Reply:
As you say, the `provider_type` is used in the `init_provider`, to match the class name in the factory pattern.
Later, it's not used at all until the `Provider`get_extended_data` method is used to complement the `Maintenance` data.
The user case would be.
1. The client (Plugin) uses a provider named `centurylink`
2. Via its own mapping, he will use the `init_provider("lumen", extended_data={"provider_type":"centurylink"}`
3. This will return the proper Provider, but when we recieve the processed `Maintenances` the `provider` will be `centurylink` instead of the default `lumen`
Obviously, this is something that you could overwrite when the object is received, so not 100% sure if it's worth the change...

Reply:
Yeah, this is feeling like a solution in search of a problem to me... I'm not yet convinced this is necessary or desirable, seems like it'll just be a source of confusion.